src/main/java/com/devonfw/ide/sonarqube/common/impl/check/naming/DevonNamingConventionInterfaceExtendsInterfaceCheck.java

```diff
@@ -0,0 +1,123 @@
+package com.devonfw.ide.sonarqube.common.impl.check.naming;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.tree.ClassTree;
+import org.sonar.plugins.java.api.tree.CompilationUnitTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.TypeTree;
+
+/**
+ * Abstract base class for naming convention checks of interfaces
+ */
+public abstract class DevonNamingConventionInterfaceExtendsInterfaceCheck implements JavaFileScanner {
+
+  /**
+   * This needs to be the suffix of the checked interface if it extends certain other interfaces.
+   */
+  protected final Pattern extendingInterfaceSuffixRegEx;
+
+  /**
+   * The constructor.
+   *
+   * @param extendingInterfaceSuffix See JavaDoc on variable declaration.
+   */
+  public DevonNamingConventionInterfaceExtendsInterfaceCheck(String extendingInterfaceSuffix) {
+
+    this.extendingInterfaceSuffixRegEx = Pattern.compile(extendingInterfaceSuffix);
+  }
+
+  /**
+   * Method called after parsing and semantic analysis has been done on file.
+   *
+   * @param context Context of analysis containing the parsed tree.
+   */
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+
+    ClassTree tree = getTreeInstance(context);
+    String interfaceName = tree.simpleName().name();
+    Set<String> superInterfacesNames = getSuperInterfacesNames(tree);
+
+    if (doesSuperInterfaceHaveRegEx(superInterfacesNames) && !doesInterfaceHaveRegEx(interfaceName)) {
+      context.reportIssue(this, tree,
+          "If a superinterface has " + this.extendingInterfaceSuffixRegEx.toString()
+              + " as suffix, then the subinterface should also have " + this.extendingInterfaceSuffixRegEx.toString()
+              + " as suffix");
+      // context.addIssueOnFile(this,
+      // "If a superinterface has " + this.extendingInterfaceSuffixRegEx.toString()
+      // + " as suffix, then the subinterface should also have " + this.extendingInterfaceSuffixRegEx.toString()
+      // + " as suffix");
```

Comment:
for clean code you should remove outcommeted code. However, not preventing to merge the PR.